internal/discord/command-handlers.go

```diff
@@ -479,3 +479,96 @@ func (d *Discord) InteractionCreate(s *discordgo.Session, i *discordgo.Interacti
 		d.ShowAllWarnings(s, i)
 	}
 }
+
+// ExileCheckUserHelper checks whether the user meets the requirements to be exiled
+// returns nil if the user does not and returns the member on success
+func (d *Discord) ExileCheckUserHelper(state *InteractionState, userID string) (*discordgo.Member, error) {
+	// Check if user exists in guild
+	member, err := d.GetUserInGuild(state.interaction.GuildID, userID)
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not find user <@%v> in guild", userID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(state.session, state.interaction.Interaction, tempstr, &state.isFirst)
+		return nil, err
+	}
+	return member, nil
+}
+
+// ExileRoleHelper removes a role `roleIDToRemove` and then adds a role `roleIDToAdd` to the user `userID`
+func (d *Discord) ExileRoleHelper(state *InteractionState, userID string, roleIDToRemove string, roleIDToAdd string) error {
+	// Attempt to remove role first
+	err := state.session.GuildMemberRoleRemove(state.interaction.GuildID, userID, roleIDToRemove)
+	// Abort entire process if role removal fails
+	if err != nil {
+		tempstr := fmt.Sprintf("Could not remove role %v from user <@%v>", roleIDToAdd, userID)
+		fmt.Printf("%v: %v\n", tempstr, err)
+		RespondToInteraction(state.session, state.interaction.Interaction, tempstr, &state.isFirst)
+		AppendLogMsgDescription(state.logMsg, tempstr)
+		return err
+	} else {
+		// Add role
```

Comment:
likewise, check to see if they already have the role. we should probably move the checks to a helper and call that helper at the beginning of the func